src/LiveQuery/RoleCache.js

```diff
@@ -0,0 +1,48 @@
+import Parse from 'parse/node';
+import LRU from 'lru-cache';
+import logger from '../logger';
+
+function rolesForUserId(userId) {
+  const user = new Parse.User();
+  user.id  = userId;
+  const rolesQuery = new Parse.Query(Parse.Role);
+  rolesQuery.equalTo("users", user);
+  return rolesQuery.find({useMasterKey:true});
```

Comment:
This is invalid, as roles can be referenced in roles etc...

Reply:
Also, the roles cache should /could be shared with the existing parse-server one